### PR TITLE
Make JSON logging optional

### PR DESCRIPTION
**Pull request by gp510**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -164,6 +164,7 @@ SPLUNK_LOGGING_DRIVER_BUFFER_MAX | The maximum amount of messages to hold in buf
 SPLUNK_LOGGING_DRIVER_CHANNEL_SIZE | How many pending messages can be in the channel used to send messages to background logger worker, which batches them. | 4 * 1000
 SPLUNK_LOGGING_DRIVER_TEMP_MESSAGES_HOLD_DURATION | Appends logs that are chunked by docker with 16kb limit. It specifies how long the system can wait for the next message to come. | 100ms 
 SPLUNK_LOGGING_DRIVER_TEMP_MESSAGES_BUFFER_SIZE	| Appends logs that are chunked by docker with 16kb limit. It specifies the biggest message in bytes that the system can reassemble. The value provided here should be smaller than or equal to the Splunk HEC limit. 1 MB is the default HEC setting. | 1048576 (1mb)
+SPLUNK_LOGGING_DRIVER_JSON_LOGS	| Determines if JSON logging is enabled. | true
```
Comment (dtregonning) [inline README.md]: Can we provide more information on what JSON Logging here in the readme. Currently assumes knowledge. A link to docker docs is fine.

File: message_processor.go [hunk 2]
```diff
@@ -45,6 +48,12 @@ This is a routine to decode the log stream into LogEntry and store it in buffer
 and send the buffer to splunk logger and json logger
 */
 func (mg messageProcessor) consumeLog(lf *logPair) {
+	// Check env variable for json logs
+	jsonLogs, err := strconv.ParseBool(os.Getenv(envVarJSONLogs))
```
Comment (dtregonning) [inline message_processor.go]: are we pulling this value from an OS environment variable or are we setting up a switch from a param? I'd follow the same way we do buffer variable set/get.

File: message_processor.go [hunk 3]
```diff
@@ -86,9 +95,11 @@ func (mg messageProcessor) consumeLog(lf *logPair) {
 			}
 			// Append to temp buffer
 			if err := tmpBuf.append(&buf); err == nil {
-				// Send message to splunk and json logger
+				// Send message to splunk and also json logger if enabled
 				mg.sendMessage(lf.splunkl, &buf, tmpBuf, lf.info.ContainerID)
-				mg.sendMessage(lf.jsonl, &buf, tmpBuf, lf.info.ContainerID)
+				if jsonLogs {
```
Comment (dtregonning) [inline message_processor.go:95]: logic here looks fine

File: message_processor.go [hunk 4]
```diff
@@ -31,6 +32,8 @@ import (
 	protoio "github.com/gogo/protobuf/io"
 )
 
+const envVarJSONLogs = "SPLUNK_LOGGING_DRIVER_JSON_LOGS"
```
Comment (dtregonning) [inline message_processor.go]: all param setup and consts are set up in splunk_logger.go
Follow same methodology for adding this param

**Review by dtregonning — CHANGES_REQUESTED** [review]
Hey @gp510 - added some changes needed to get this through. Good start.

File: README.md (hunk 1 above)
Comment (gp510) [reply README.md]: Link provided

File: message_processor.go (hunk 2 above)
Comment (gp510) [reply message_processor.go]: OS environment, same as buffer settings

File: message_processor.go (hunk 4 above)
Comment (gp510) [reply message_processor.go]: Following same pattern of defining the constant locally

Comment (dtregonning) [reply message_processor.go]: sorry thats not the case @gp510 - move const to splunk_logger.go its where all parameter constants are defined.

File: message_processor.go (hunk 2 above)
Comment (dtregonning) [reply message_processor.go]: This seems off we should be using parameters read in to plugin. The buffer does not use os.Getenv to retrieve this value. just reference variable direct

`jsonLogs, err := strconv.ParseBool(envVarJSONLogs)`

File: message_processor.go [hunk 5]
```diff
@@ -45,6 +48,12 @@ This is a routine to decode the log stream into LogEntry and store it in buffer
 and send the buffer to splunk logger and json logger
 */
 func (mg messageProcessor) consumeLog(lf *logPair) {
+	// Check env variable for json logs
```
Comment (dtregonning) [inline message_processor.go]: modify comment - this should come from config not from os variable

**Review by dtregonning — CHANGES_REQUESTED** [review]
Hey @gp510 still needs some tweaks

File: message_processor.go (hunk 2 above)
Comment (gp510) [reply message_processor.go]: Created getAdvancedOptionBool function in splunk_logger.go to handle jsonLogs

File: message_processor.go (hunk 4 above)
Comment (gp510) [reply message_processor.go]: Now moves to splunk_logger.go

File: message_processor.go (hunk 5 above)
Comment (gp510) [reply message_processor.go]: Comment removed

File: splunk_logger.go [hunk 6]
```diff
@@ -437,6 +440,19 @@ func getAdvancedOptionInt(envName string, defaultValue int) int {
...
 }
 
+func getAdvancedOptionBool(envName string, defaultValue bool) bool {
+	valueStr := os.Getenv(envName)
+	if valueStr == "" {
+		return defaultValue
+	}
+	parsedValue, err := strconv.ParseBool(valueStr)
+	if err != nil {
+		logrus.Error(fmt.Sprintf("Failed to parse value of %s as integer. Using default %v. %v", envName, defaultValue, err))
```
Comment (dtregonning) [inline splunk_logger.go]: failed to parse value as boolean - update log message

**Review by dtregonning — APPROVED** [review]
Make one small update to log message - and if all tested - please merge.
Thanks @gp510 - good one

**Pull request merged**